meta-lmp-base/recipes-multimedia/alsa/alsa-utils_%.bbappend

```diff
@@ -0,0 +1,11 @@
+# Don't use /var for ostree-based distro to store/look for "asound.state"
+# file which is required for some sound systems (i.e. wm8960 on imx8mp-evk)
+# to work properly.
+
+python __anonymous() {
+    if bb.utils.contains('IMAGE_CLASSES', 'image_types_ostree', True, False, d):
+        d.setVar("localstatedir", d.getVar("prefix"))
+}
+
+EXTRA_OECONF:append = " --with-asound-state-dir=${localstatedir}/lib/alsa"
+EXTRA_OEMAKE:append = " SYS_ASOUND_DIR=${localstatedir}/lib/alsa"
```

Comment:
One problem I see is that this is also used by alsactl when saving custom mixer data, and that will fail when stored in /usr/lib.

I think it is ok to have it using /var/lib/alsa, as long the directory is created at runtime via tmpfiles.d and that the default asound.state file can be consumed from another location (or if that file gets copied over together with the process that is responsible for creating the folder at runtime).

Then we can have the same solution for both lmp-base and lmp.

Reply:
I need to test this variant. The question is where alsactl is called first, from udev rules without path to the asound.state or from post-install package script, where we alter the path to the config file during image building.

Reply:
so on all devices that have sound (in this case imx boards) we should enable alsa by default or make it a MACHINE_FEATURE

Reply:
@ricardosalveti I changed the solution preserving default behavior.

Reply:
Do we still need "--with-asound-state-dir=${localstatedir}/lib/alsa" and "SYS_ASOUND_DIR=${localstatedir}/lib/alsa"?

Reply:
> Do we still need "--with-asound-state-dir=${localstatedir}/lib/alsa" and "SYS_ASOUND_DIR=${localstatedir}/lib/alsa"?

Not mandatory, but what if we build a distro with a variable directory other than /var? :)

Reply:
Then we will have way bigger problems :D

Reply:
Ok, got rid that stuff :)